moderation/commands.go

```diff
@@ -156,6 +156,52 @@ var ModerationCommands = []*commands.YAGCommand{
 			return GenericCmdResp(MABanned, target, parsed.Switch("d").Value.(time.Duration), true, false), nil
 		},
 	},
+	&commands.YAGCommand{
+		CustomEnabled: true,
+		CmdCategory:   commands.CategoryModeration,
+		Name:          "Unban",
+		Aliases:       []string{"unbanid"},
+		Description:   "Unbans a user. Reason requirement is same as ban command setting.",
+		RequiredArgs:  1,
+		Arguments: []*dcmd.ArgDef{
+			&dcmd.ArgDef{Name: "User", Type: dcmd.UserID},
+			&dcmd.ArgDef{Name: "Reason", Type: dcmd.String},
+		},
+		
+		RunFunc: func(parsed *dcmd.Data) (interface{}, error) {
+			config, _, err := MBaseCmd(parsed, 0) //in most situations, the target will not be a part of server, hence no point in doing unnecessary api calls(i.e. bot.GetMember)
+			if err != nil {
+				return nil, err
+			}
+
+			reason := SafeArgString(parsed, 1)
+			reason, err = MBaseCmdSecond(parsed, reason, config.BanReasonOptional, discordgo.PermissionBanMembers, config.BanCmdRoles, config.BanEnabled)
+			if err != nil {
+				return nil, err
+			}
+			targetID := parsed.Args[0].Int64()
+			target := &discordgo.User{
+					Username:      "unknown",
+					Discriminator: "????",
+					ID:            targetID,
+				  }
+			targetMem := parsed.GS.MemberCopy(true, targetID)
```

Comment:
Here the intention was to prevent unnecessary api call in case member already exists in state. getMember falls back to fetching member from api if not in state. I decided to proceed with unban attempt if member is not found in state and then handle error to see if member was indeed banned. Let me know if you think it woukd be still advisable to use getMember here.

Reply:
this is gonna be such a cold code path that there's no point in saving a api call here, api calls in commands is cheap because overall less than 10 commands per second is ran.

Reply:
+ I have plans on adding querying of members over the gateway so then it wouldn't even be a "api" call per se

Reply:
Gotcha! changed to bot.GetMember 👍